### PR TITLE
Fix(eos_validate_state): ANTA Fix AvdTestInbandReachability to support the new inband management data model

### LINKED ISSUE #3404: Fix(eos_validate_state): ANTA AvdTestInbandReachability should be refactored to support the new AVD 4.X inband management data models
The current implementation of AvdTestInbandReachability follows the legacy eos_validate_state; we test Loopback0 reachability from `management_interfaces`  when `type: inband`.  Since AVD 4.X, inband management interfaces can be ethernet, loopback or vlan interfaces with `type: inband_mgmt`.

AvdTestInbandReachability test should be refactored to support the new data models.

### PR DESCRIPTION
**Pull request by carl-baillargeon**
## Change Summary

Fix `AvdTestInbandReachability` to support the new AVD 4.X inband management data models. Also updated molecule to properly test all tests (including optional ones).

## Related Issue(s)

Fixes #3404

## Component(s) name

`arista.avd.eos_validate_state` **ANTA Mode**

## Proposed changes

- `AvdTestInbandReachability` will now check all `vlan_interfaces` to find `type: inband_mgmt` and test reachability from this interface to all Loopback0 of other devices in the fabric.
- Add support for a special tag `run_all` to run all tests from Ansible. This is useful to run all tests, including optional tests (in molecule as an example).

## How to test
See molecule

### User Checklist

- N/A

### Repository Checklist

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [X] My change requires a change to the documentation and documentation have been updated accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/get_anta_results.py [hunk 1]
```diff
@@ -149,6 +149,7 @@ def get_skipped_tests_from_tags(run_tags: tuple, skip_tags: tuple) -> list[dict]
       result (list[dict]): List of dictionary containing the categories to skip.
     """
     result = []
+
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/get_anta_results.py]: is this change expected?

File: ansible_collections/arista/avd/molecule/eos_validate_state/inventory/group_vars/DC1.yml [hunk 2]
```diff
@@ -117,14 +117,17 @@ l2leaf:
   defaults:
     platform: vEOS-lab
     spanning_tree_mode: mstp
-
+    inband_mgmt_subnet: 172.21.110.0/24
+    inband_mgmt_description: L2LEAF_INBAND_MGMT
+    inband_mgmt_vlan_name: L2LEAF_INBAND_MGMT
+    inband_mgmt_vlan: 4085
+    inband_mgmt_vrf: MGMT
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/molecule/eos_validate_state/inventory/group_vars/DC1.yml:206]: Can we do same for l3 leaf switches, i was trying it but then am not getting inband_mgmt_vlan 4085 in structured  config.

Comment (carl-baillargeon) [reply ansible_collections/arista/avd/molecule/eos_validate_state/inventory/group_vars/DC1.yml:206]: The `eos_designs` data model is for L2 switches --> https://avd.sh/en/stable/roles/eos_designs/docs/input-variables.html#node-type-inband-management

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/get_anta_results.py (hunk 1 above)
Comment (carl-baillargeon) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/get_anta_results.py]: No :) Fixed in c510c2ef8d7682a5c68f30e22b78542316bdd6e8

**Review by MaheshGSLAB — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**